lib/api/include/irods/replica_truncate.h

```diff
@@ -0,0 +1,82 @@
+#ifndef IRODS_REPLICA_TRUNCATE_H
+#define IRODS_REPLICA_TRUNCATE_H
+
+struct RcComm;
+struct DataObjInp;
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/// Truncate a replica for the specified data object to the specified size.
+///
+/// \parblock
+/// This API selects a replica to truncate according to the rules of POSIX truncate(2). The caller may provide keywords
+/// via condInput in order to influence the hierarchy resolution for selecting a replica to truncate.
+/// \endparblock
+///
+/// \param[in] _comm A pointer to a RcComm.
+/// \param[in] _inp \parblock
+/// DataObjInp structure which requires the following inputs:
+///     objPath - The full logical path to the target data object.
+///     dataSize - The desired size of the replica after truncating.
+///
+/// The condInput supports the following keywords for hierarchy resolution for a "write" operation:
+///     replNum - The replica number of the replica to truncate.
+///     rescName - The name of the resource with the replica to truncate. Must be a root resource.
+///     defRescName - The default resource to target in the absence of any other inputs or policy.
```

Comment:
We don't have much choice in exposing "defRescName" because this is used in `resolve_resource_hierarchy` and all APIs which resolve resource hierarchies support its usage implicitly. The default resource provided by a client environment has always been there (AFAIK) and has always been override-able by admin-configured policy in the server. I was trying to document everything I could think of in the spirit of https://github.com/irods/irods/issues/5859. That being said, there are technically other keywords used by that machinery not stated here.

Should we choose to not support it in this API, I suppose we would have to treat its presence as an error or remove it from the `condInput`. Alternatively, `replNum` or `rescName` would become required inputs. I'm not sure how I feel about that since we don't know what this API is going to be used for.

Reply:
All data-moving iCommands support it, and I believe PRC also supports it for at least one operation. Can't speak to the others.

Reply:
Just spotted a reference to the default resc keyword in putUtil.

Carry on.